app/models/plan.rb

```diff
@@ -617,6 +617,21 @@ def owner
     return nil
   end
 
+  ##
+  # returns the shared roles of a plan, excluding the creator
+  def shared
+    role_values = Role.access_values_for(:commenter)
+                      .concat(Role.access_values_for(:editor))
+                      .concat(Role.access_values_for(:administrator)).uniq! - Role.access_values_for(:creator)
+    shared_roles = []
+    self.roles.each do |role|
```

Comment:
You're not using the `role_values` you collected above efficiently! `self.roles` is going to run a DB query for every role.
Use `Role.where(plan: self, access: role_values)` instead. It will hit the DB once and return any roles that are not 'creator'. No need to loop through the roles association